### PR TITLE
Add Basic Logic Building Functions in Go

### PR DESCRIPTION
**Pull request by pasivinay**
### **Description:
This PR introduces fundamental Go functions for logic-building exercises, covering loops, conditionals, and arithmetic operations. Test cases are included for validation.**

**Changes:**

Implemented functions:

> 1. EvenOdd – Checks if a number is even or odd.
> 2. Table – Generates a multiplication table.
> 3. Sum – Computes the sum of the first n natural numbers.
> 4. Swap – Swaps two integers.
> 5. ClosestNumber – Finds the closest number to n that is divisible by m.
> 6. OppOfDice – Returns the opposite face of a six-sided die.
> 7. NthTermOfAP – Calculates the nth term of an arithmetic progression.

Testing:

> 1. Added unit tests for all functions.
> 2. Verified edge cases and expected outputs.

### CODE REVIEW

File: exercises/logic_building/basics/basics.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package basics
+
+import (
+	"fmt"
+	"math"
+)
+
+// This function takes a int and returns whether it's even(true) or odd(false).
+
```
Comment (justsushant) [inline exercises/logic_building/basics/basics.go]: Lets not keep line gaps b/ween function declaration & its associated comment. 
For two reasons:
- Comments won't show up as the part of documentation if you hover on the function in VSCode
- Unnecessary line gaps just look ugly

Comment (pasivinay) [reply exercises/logic_building/basics/basics.go]: Fix:

- The gaps from comments have been removed.
- Also added function names in the beginning of the comments describing the function.

File: exercises/logic_building/basics/basics.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	// swapping numbers
+	a = b
+	b = c
+
+	return a, b
+}
+
+// abs takes two integers and find the number closest to n and divisible by m.
+func abs(a int) int {
+	return -a
```
Comment (justsushant) [inline exercises/logic_building/basics/basics.go]: Please recheck the implementation of the function. Feels wrong to me.

File: exercises/logic_building/basics/basics_test.go [hunk 3]
```diff
@@ -0,0 +1,138 @@
+package basics
+
+import (
+	"testing"
+)
+
+func TestEvenOdd(t *testing.T) {
+	t.Run("Check whether the 4 is even or odd", func(t *testing.T) {
```
Comment (justsushant) [inline exercises/logic_building/basics/basics_test.go]: Subtests are nice, but table driven tests are nicer. Please refactor all these tests into table driven tests.
Please take inspiration from here: https://go.dev/wiki/TableDrivenTests

Comment (pasivinay) [reply exercises/logic_building/basics/basics_test.go]: Added table driven tests for all functions.

File: exercises/logic_building/basics/basics.go (hunk 2 above)
Comment (pasivinay) [reply exercises/logic_building/basics/basics.go]: Fixed the implementation of abs function.
